tests/TestResultsTests/ReentryTests.ipf

```diff
@@ -0,0 +1,60 @@
+#pragma TextEncoding = "UTF-8"
+#pragma rtGlobals=3				// Use modern global access method and strict wave access
+#pragma DefaultTab={3,20,4}		// Set default tab width in Igor Pro 9 and later
+#pragma version=1.09
+#pragma ModuleName = TS_ReentryTests
+
+#include "unit-testing"
+#include "TestUtils"
+
+// These are tests that check if UTF works correctly with re-entry tests
+
+// a simple background task that directly exists after its first run
+static Function ReEntryTask(s)
+	STRUCT WMBackgroundStruct &s
+
+	PASS()
```

Comment:
Do we officially support assertions in background tasks now?

Reply:
There is nothing that would stop you. And with the current changes it is possible to do so as the execution of UTF is halted as if it is still inside the testing method.

And background tasks do still run in the main thread so the global state of UTF is accessible.

Reply:
Do you have a scenario where supporting assertions in background tasks can lead to some bugs?

Reply:
> Do we officially support assertions in background tasks now?

I don't think we officially say that this works, but I also don't see an issue.

Reply:
Yes we are still within the testing framework, thus in that regard the background task that runs can issue assertions.
The thought here is that typically the background task that is monitored is part of the code that is tested and not part of the test case. 
If there would be an application case where the test case creates the background task, we should mention that as official feature. Though, beside testing the testing framework I currently don't have an idea where I would need that.

I am ok with keeping it unofficial.